### PR TITLE
[FEATURE] add default save button to user profile (Podio bug_74)

### PR DESCRIPTION
**Pull request by yosokus**

### CODE REVIEW

File: Classes/Xclass/DefaultSplitButton.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        $user = $this->getBackendUser();
+        if (isset($user->user['default_save_button']) && ! empty($user->user['default_save_button'])) {
+            $defaultButtonAction = $user->user['default_save_button'];
+        } else {
+            return parent::getButton();
+        }
+
+        /** @var InputButton $primaryAction */
```
Comment (laurisaarni) [inline Classes/Xclass/DefaultSplitButton.php]: @yosokus at this point of the code the phpdoc comment is not valid. The variable is defined null. Move the comment to proper context.

File: Classes/Xclass/DefaultSplitButton.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    protected $defaultButtonSet = FALSE;
+
+    /**
+     * Returns the current button
+     *
+     * @return array
+     */
+    public function getButton() {
+        if ($this->defaultButtonSet) {
+            return $this->items;
```
Comment (laurisaarni) [inline Classes/Xclass/DefaultSplitButton.php]: @yosokus the flow of the code is not very clear. Multiple return statements all over the function. Please follow clean code practices.

**Comment by yosokus** [comment]
@laurisaarni please review the new pull request

**Comment by laurisaarni** [comment]
@yosokus it's OK.

**Comment by dmh** [comment]
Task Description

```
Save and close should be remmebered as a choice

Right now it always falls back to default Save only. This does not suit most of the users.
Can be also an option in the user settings to override the default button.
```

I would like to use an additional extension for such feature. What do think about it guys?
https://typo3.org/extensions/repository/?id=23&L=0&q=save+button

@MattiasNilsson @jozefspisiak

**Comment by dmh** [comment]
close PR, because no answer for a long time

**Pull request closed without merging**